src/doc/en/developer/doctesting.rst

```diff
@@ -1338,9 +1343,267 @@ To give a json file storing the timings for each file, use the
+in detail in :ref:`section-modularized-doctesting`.  Our examples in this section
+refer to this setting, but it applies the same to any user-created virtual environments.
+
+The virtual environments, set up in directories such as
+``pkgs/sagemath-standard/.tox/sagepython-sagewheels-nopypi-norequirements``
+contain installations of built (non-editable) wheels.
+
+To test all modules of Sage that are installed in a virtual environment,
+use the option ``--installed`` (instead of ``--all``)::
+
+    [mkoeppe@sage sage]$ pkgs/sagemath-standard/.tox/sagepython-sagewheels-.../sage -t \
+                             -p4 --installed
+
+This tests against the doctests as they appear in the installed copies of the files
+(in ``site-packages/sage/...``).
+Note that these installed copies should never be edited, as they can
+be overwritten without warning.
+
+When testing a modularized distribution package other than sagemath-standard,
+the top-level module :mod:`sage.all` is not available.  Use the option ``--environment``
+to select an appropriate top-level module::
+
+    [mkoeppe@sage sage]$ pkgs/sagemath-categories/.tox/sagepython-sagewheels-.../sage -t \
+                             -p4 --environment sage.all__sagemath_categories             \
+                             --installed
+
+To test the installed modules against the doctests as they appear in the source
+tree (``src/sage/...``)::
+
+    [mkoeppe@sage sage]$ pkgs/sagemath-categories/.tox/sagepython-sagewheels-.../sage -t \
+                             -p4 --environment sage.all__sagemath_categories             \
+                             src/sage/structure
+
+Note that testing all doctests as they appear in the source tree does not make sense
+because many of the source files may not be installed in the virtual environment.
+Use the option ``--only-lib`` to skip the source files of all Python/Cython modules
+that are not installed in the virtual environment::
+
+    [mkoeppe@sage sage]$ pkgs/sagemath-categories/.tox/sagepython-sagewheels-.../sage -t \
+                             -p4 --environment sage.all__sagemath_categories --only-lib  \
+                             src/sage/schemes
+
+This option can also be combined with ``--all``::
+
+    [mkoeppe@sage sage]$ pkgs/sagemath-categories/.tox/sagepython-sagewheels-.../sage -t \
+                             -p4 --environment sage.all__sagemath_categories --only-lib  \
+                             --all
+
+
+.. _section-fixdoctests:
+
+The doctest fixer
+=================
+
+Sage provides a development tool that assists with updating doctests.
+
+
+Updating doctest outputs
+------------------------
+
+By default, ``./sage --fixdoctests`` runs the doctester and replaces the expected outputs
+of all examples by the actual outputs from the current version of Sage::
+
+    [mkoeppe@sage sage]$ ./sage --fixdoctests \
+                                --overwrite src/sage/arith/weird.py
+
```

Comment:
Yes, it's redundant here. This flag is only needed if exactly 2 files are given, but I use `--overwrite` in all examples so that users do not need to be aware of this quirk of the interfaces.

Actually, I would like to deprecate the special case of 2 file arguments; thoughts?

Reply:
> Yes, it's redundant here. This flag is only needed if exactly 2 files are given, but I use `--overwrite` in all examples so that users do not need to be aware of this quirk of the interfaces.

The most frequent case would be 1 file input case. As all your examples are accompanied with `--overwrite` or `--no-overwrite`, it makes an impression that `--(no)-overwrite` is always needed. 

After the first example with no flags is given, then you may mention that `--overwrite` is implied there.

> 
> Actually, I would like to deprecate the special case of 2 file arguments; thoughts?

I agree.

Reply:
OK.

Reply:
> > Actually, I would like to deprecate the special case of 2 file arguments; thoughts?
> 
> I agree.

Done in 3c44670ef5